impl/src/main/java/org/ehcache/internal/store/disk/DiskStore.java

```diff
@@ -1041,9 +1037,18 @@ public void remove() {
       Serializer<Element> elementSerializer = serializationProvider.createSerializer(Element.class, storeConfig.getClassLoader());
       Serializer<Object> objectSerializer = serializationProvider.createSerializer(Object.class, storeConfig.getClassLoader());
       
+      // todo: This should be enforced at the type system
+      Object identifier;
+      if(storeConfig instanceof PersistentStoreConfiguration<?, ?, ?>) {
+        identifier = ((PersistentStoreConfiguration) storeConfig).getIdentifier();
+      } else {
+        throw new IllegalArgumentException("Store.Configuration for DiskStore should implement Store.PersistentStoreConfiguration");
```

Comment:
No. We need to know where to write the data. @ljacomet also found the meaning of `Persistence` to be somewhat conflated... not sure if it is. My take is temp_swap (which btw doesn't exist, I realize I hope we're not naming it that) Is merely lifecycling the data of the cache, i.e. delete on exit, it still is using "persistence"...